src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java

```diff
@@ -0,0 +1,584 @@
+     * @return String of errors, or an empty string if there are no errors found.
+     */
+    private String getErrors(JsonNode jsonResponse) {
+        if (jsonResponse.has("error")) {
+            StringBuilder errorOutput = new StringBuilder();
+            ArrayNode errors = (ArrayNode) jsonResponse.at(ERROR_URI);
+            for (int i = 0; i < errors.size(); i += 2) {
+                if (errorOutput.length() > 0)
+                    errorOutput.append(", ");
+                errorOutput.append("Error Code: ").append(errors.get(i).asText());
+                if ((i + 1) < errors.size()) {
+                    errorOutput.append("- ").append(errors.get(i + 1).asText());
+                }
+            }
+            return errorOutput.toString();
+        }
+        return "";
+    }
+
+    @Override
+    public void controlProperty(ControllableProperty cp) throws Exception {
+        String prop = cp.getProperty();
+        String value = String.valueOf(cp.getValue());
+
+        if (prop.endsWith(VOLUME)) {
+            String target = prop.substring(8, prop.length() - 7);
+            this.doPost("audio", setAudioVolume.withParams(ImmutableMap.of("volume", value, target, target)));
+            return;
+        } else if (prop.endsWith(MUTE)) {
+            this.doPost("audio", setAudioMute.withParams(ImmutableMap.of("status", value.equals("1"))));
+            return;
+        }
+
+        switch (prop) {
+            case SOUND_OUTPUT_TERMINAL:
+                this.doPost(AUDIO_URI, setSoundSettings.withParams(ImmutableMap.of("settings",
+                        ImmutableMap.of("value", value, "target", OUTPUT_TERMINAL))));
+                break;
+            case SPEAKER_TV_POSITION:
+                this.doPost(AUDIO_URI, setSpeakerSettings.withParams(ImmutableMap.of("settings",
+                        ImmutableMap.of("value", value, "target", TV_POSITION))));
+                break;
+            case SPEAKER_SUBWOOFER_LEVEL:
+                this.doPost(AUDIO_URI, setSpeakerSettings.withParams(ImmutableMap.of("settings",
+                        ImmutableMap.of("value", value, "target", SUBWOOFER_LEVEL))));
+                break;
+            case SPEAKER_SUBWOOFER_FREQ:
+                this.doPost(AUDIO_URI, setSpeakerSettings.withParams(ImmutableMap.of("settings",
+                        ImmutableMap.of("value", value, "target", SUBWOOFER_FREQ))));
+                break;
+            case SPEAKER_SUBWOOFER_PHASE:
+                this.doPost(AUDIO_URI, setSpeakerSettings.withParams(ImmutableMap.of("settings",
+                        ImmutableMap.of("value", value, "target", SUBWOOFER_PHASE))));
+                break;
+            case SPEAKER_SUBWOOFER_POWER:
+                this.doPost(AUDIO_URI, setSpeakerSettings.withParams(ImmutableMap.of("settings",
+                        ImmutableMap.of("value", "0".equals(value) ? "off" : "on", "target", SUBWOOFER_POWER))));
+                break;
```

Comment:
Looks like `AUDIO_URI` is used for any sound related items...ignore my previous comment :)